pkg/jsonlog/jsonlog_test.go

```diff
@@ -12,29 +13,122 @@ import (
+// Invalid json should return an error
+func TestWriteLogWithInvalidJSON(t *testing.T) {
+	json := strings.NewReader("Invalid json")
+	w := bytes.NewBuffer(nil)
+	if err := WriteLog(json, w, "json", time.Time{}); err == nil {
+		t.Fatalf("Expected an error, got [%v]", w.String())
+	}
+}
+
+// Any format is valid, it will just print it
+func TestWriteLogWithInvalidFormat(t *testing.T) {
+	testLine := "Line that thinks that it is log line from docker\n"
 	var buf bytes.Buffer
 	e := json.NewEncoder(&buf)
+	for i := 0; i < 35; i++ {
+		e.Encode(JSONLog{Log: testLine, Stream: "stdout", Created: time.Now()})
+	}
+	w := bytes.NewBuffer(nil)
+	if err := WriteLog(&buf, w, "invalid format", time.Time{}); err != nil {
+		t.Fatal(err)
+	}
+	res := w.String()
+	t.Logf("Result of WriteLog: %q", res)
+	lines := strings.Split(strings.TrimSpace(res), "\n")
+	expression := "^invalid format Line that thinks that it is log line from docker$"
+	logRe := regexp.MustCompile(expression)
+	expectedLines := 35
+	if len(lines) != expectedLines {
+		t.Fatalf("Must be %v lines but got %d", expectedLines, len(lines))
+	}
+	for _, l := range lines {
+		if !logRe.MatchString(l) {
+			t.Fatalf("Log line not in expected format [%v]: %q", expression, l)
+		}
+	}
+}
+
+// Having multiple Log/Stream element
+func TestWriteLogWithMultipleStreamLog(t *testing.T) {
 	testLine := "Line that thinks that it is log line from docker\n"
-	for i := 0; i < 30; i++ {
+	var buf bytes.Buffer
+	e := json.NewEncoder(&buf)
+	for i := 0; i < 35; i++ {
 		e.Encode(JSONLog{Log: testLine, Stream: "stdout", Created: time.Now()})
 	}
 	w := bytes.NewBuffer(nil)
-	format := timeutils.RFC3339NanoFixed
-	if err := WriteLog(&buf, w, format, time.Time{}); err != nil {
+	if err := WriteLog(&buf, w, "invalid format", time.Time{}); err != nil {
 		t.Fatal(err)
 	}
 	res := w.String()
 	t.Logf("Result of WriteLog: %q", res)
 	lines := strings.Split(strings.TrimSpace(res), "\n")
-	if len(lines) != 30 {
-		t.Fatalf("Must be 30 lines but got %d", len(lines))
+	expression := "^invalid format Line that thinks that it is log line from docker$"
+	logRe := regexp.MustCompile(expression)
+	expectedLines := 35
+	if len(lines) != expectedLines {
+		t.Fatalf("Must be %v lines but got %d", expectedLines, len(lines))
 	}
-	// 30+ symbols, five more can come from system timezone
-	logRe := regexp.MustCompile(`.{30,} Line that thinks that it is log line from docker`)
 	for _, l := range lines {
 		if !logRe.MatchString(l) {
-			t.Fatalf("Log line not in expected format: %q", l)
+			t.Fatalf("Log line not in expected format [%v]: %q", expression, l)
+		}
+	}
+}
+
+// Write log with since after created, it won't print anything
+func TestWriteLogWithDate(t *testing.T) {
+	created, _ := time.Parse("YYYY-MM-dd", "2015-01-01")
+	var buf bytes.Buffer
+	testLine := "Line that thinks that it is log line from docker\n"
+	jsonLog := JSONLog{Log: testLine, Stream: "stdout", Created: created}
+	e := json.NewEncoder(&buf)
+	e.Encode(jsonLog)
```

Comment:
super tiny nit that you can just ignore, I'd prefer `err := ...NewEncoder(&buf).Encode(jsonLog)` and check error :)